lib/sampling.h

```diff
@@ -5,17 +5,18 @@
 #include "xorshift.h"
 
 #include <cmath>
+#include <thread>
 #include <tuple>
 
 namespace {
     static constexpr float M_2PI = 2.f * M_PI;
 }
 
+namespace sampling {
 
-class Hemisphere {
-public:
-    // TODO: Docs
-    std::pair<Vec, float> sample() {
+    xorshift64star<float> uniform{4};
```

Comment:
So, we leave as is? Not thread_local? Then we should make it static, to hide the variable. (!) static here means something completely different, than static for variables inside a function.